### DISCUSSION TITLE
Error setting fps

### DISCUSSION THREAD

**Discussion by Loeschen** [Motion-Project/motion#1722] [Q&A]
Hello, unfortunately I have the following errors in the log.  My start parameter is `libcamerify motion`

```
[1:ml1] [NTC] [VID] [Nov 03 22:54:00] v4l2_pixfmt_select: Selected palette Y12 
[1:ml1] [ERR] [VID] [Nov 03 22:54:00] v4l2_fps_set: Error setting fps. Return code -1
[1:ml1] [ERR] [VID] [Nov 03 22:54:00] v4l2_mmap_set: Error starting stream. VIDIOC_STREAMON: Invalid argument
[1:ml1] [ERR] [VID] [Nov 03 22:54:00] vid_start: V4L2 device failed to open
```

I hope you can help me, unfortunately I can't find a solution.

**Comment by Mr-Dave** [comment]
First try to restart your PI.  Next, if it is running the 64bit raspberry PI os, you'll need to either build Motion from source or downgrade to the 32 bit version until the next Motion release.  (There was issue #1644 that caused libcamerify and Motion to fail on 64bit systems)

**Comment by Loeschen** [comment]
I have now installed "Raspberry Pi OS (Legacy) Lite 32-bit (bullseye)" and have installed the "pi_bullseye_motion_4.5.1-1", but I still get the following error.


```
[1:ml1] [NTC] [VID] v4l2_pixfmt_try: Testing palette Y12  (640x480)
[1:ml1] [NTC] [VID] v4l2_pixfmt_set: Using palette Y12  (640x480)
[1:ml1] [NTC] [VID] v4l2_pixfmt_select: Selected palette Y12
[1:ml1] [ERR] [VID] v4l2_fps_set: Error setting fps. Return code -1
[1:ml1] [ERR] [VID] v4l2_mmap_set: Error starting stream. VIDIOC_STREAMON: Invalid argument
[1:ml1] [ERR] [VID] vid_start: V4L2 device failed to open

```

I have no idea what I'm doing wrong :/

**Comment by Mr-Dave** [comment]
I'm not sure.  The PI camera does work with `libcamerify motion` so you may want to validate it works with other libcamera applications

**Reply by Loeschen** [reply]
`libcamera-hello` works without any problems. I think it is due to the camera. The Raspberry Pi Camera Module 3 is probably also causing problems for other people.

**Reply by Mr-Dave** [reply]
FYI.  I've not heard that anyone else has an issue with the camera module 3, libcamerify and Motion.  If there are multiple /dev/video devices, you may want to try the other ones.

**Reply by radzio** [reply]
@Loeschen have you managed to fix your problem? I have the exact same issue and I didn't find any solution :(

**Comment by Loeschen** [comment]
> @Loeschen have you managed to fix your problem? I have the exact same issue and I didn't find any solution :(

I followed this guide and it worked.
https://lookslikematrix.de/raspberry-pi/2023/10/14/motion-bookworm.html

**Reply by PicoI2** [reply]
Thank you, I spent like 4 hours on this problem !
To summarize, run :


    sudo apt install libcamera-tools libcamera-v4l2 motion -y


Then run motion with libcamerify


    sudo libcamerify motion

If this fix motion, then edit the service file `/lib/systemd/system/motion.service` to replace `motion` with `libcamerify motion`.
Thanks a lot !

**Reply by daphneclick** [reply]
>  replace `motion` with `libcamerify motion`

Ahh thanks for this. Using `sudo libcamerify motion` to test it worked, but now I'm not sure which line of motion.service file you mean? Should I be replacing `/var/log/motion` with `/var/log/libcamerify motion` ?

Edit: Totally forgot I could just try to look through that original link. I just need to replace the line
`ExecStart=/usr/bin/motion`
with
`ExecStart=/usr/bin/libcamerify /usr/bin/motion`
